tests/utils/test_patterns.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+from poetry.utils import patterns
+
+
+@pytest.mark.parametrize(
+    ["filename", "expected"],
+    [
+        (
+            "markdown_captions-2-py3-none-any.whl",
+            {
+                "namever": "markdown_captions-2",
+                "name": "markdown_captions",
+                "ver": "2",
+                "build": None,
+                "pyver": "py3",
+                "abi": "none",
+                "plat": "any",
+            },
+        ),
+        (
+            "SQLAlchemy-1.3.20-cp27-cp27mu-manylinux2010_x86_64.whl",
+            {
+                "namever": "SQLAlchemy-1.3.20",
+                "name": "SQLAlchemy",
+                "ver": "1.3.20",
+                "build": None,
+                "pyver": "cp27",
+                "abi": "cp27mu",
+                "plat": "manylinux2010_x86_64",
+            },
+        ),
+    ],
+)
+def test_wheel_file_re(filename, expected):
```

Comment:
```suggestion
def test_wheel_file_re(filename, namever, name, ver, build, pyver, abi, plat):
```

Reply:
updated in #3460

Reply:
Hey @abn, sorry for the late response. These are a lot of parameter. So in my opinion, the definition of them is harder to read.

Reply:
I am nto sure if I agree with that per-say. However, my concern is more towards maintanability and the suggestion was based onthe fact that each key were explicitly tested. The suggestion is merely a recommendation happy for this to go without it if you feel strongly about it.

Reply:
Changed as discussed at discord.

Reply:
Can we get this merged? Poetry completely breaks when importing these single-digit packages.

Reply:
Awesome to see this solution here. Since it looks like this has been resolved potentially by @finswimmer, are there any further unresolved items that you can find, @abn? This is a primary blocker of adoption in some of my workflows, otherwise I'm stuck monkey patching `poetry` to be able to support this with some packages in my dependency tree.